### PR TITLE
issue #4083 - handle missing patient_id claim

### PR DESCRIPTION
**Pull request by lmsurpre**
Signed-off-by: Lee Surprenant <lmsurpre@merative.com>

### CODE REVIEW

File: fhir-smart/src/main/java/org/linuxforhealth/fhir/smart/AuthzPolicyEnforcementPersistenceInterceptor.java [hunk 1]
```diff
@@ -1056,7 +1056,7 @@ private Map<ContextType, List<Scope>> getScopesFromToken(DecodedJWT jwt) throws
      */
     private Set<String> getPatientIdFromToken(DecodedJWT jwt) throws FHIRPersistenceInterceptorException {
         Claim claim = jwt.getClaim("patient_id");
-        if (claim.isNull()) {
+        if (claim.isMissing() || claim.isNull()) {
```
Comment (lmsurpre) [inline fhir-smart/src/main/java/org/linuxforhealth/fhir/smart/AuthzPolicyEnforcementPersistenceInterceptor.java:1059]: this is the fix

File: fhir-smart/src/test/java/org/linuxforhealth/fhir/smart/test/AuthzPolicyEnforcementTest.java [hunk 2]
```diff
@@ -353,6 +353,39 @@ public void testDirectPatientInteraction() throws FHIRPersistenceInterceptorExce
         }
     }
 
+    @Test
+    public void testUserInteraction() throws FHIRPersistenceInterceptorException {
```
Comment (punktilious) [inline fhir-smart/src/test/java/org/linuxforhealth/fhir/smart/test/AuthzPolicyEnforcementTest.java:357]: This claims to throw FHIRPersistenceInterceptorException but it catches it internally

Comment (lmsurpre) [reply fhir-smart/src/test/java/org/linuxforhealth/fhir/smart/test/AuthzPolicyEnforcementTest.java:357]: good catch
```suggestion
    public void testUserInteraction() {
```

File: fhir-smart/src/test/java/org/linuxforhealth/fhir/smart/test/AuthzPolicyEnforcementTest.java [hunk 3]
```diff
@@ -353,6 +353,39 @@ public void testDirectPatientInteraction() throws FHIRPersistenceInterceptorExce
...
+            fail("Patient interaction was not allowed but should have been", e);
+        }
+
+        try {
+            properties.put(FHIRPersistenceEvent.PROPNAME_RESOURCE_TYPE, "Patient");
+            properties.put(FHIRPersistenceEvent.PROPNAME_RESOURCE_ID, PATIENT_ID);
+            FHIRPersistenceEvent event = new FHIRPersistenceEvent(patient, properties);
+            interceptor.beforeRead(event);
+            interceptor.beforeVread(event);
+            interceptor.beforeHistory(event);
```
Comment (punktilious) [inline fhir-smart/src/test/java/org/linuxforhealth/fhir/smart/test/AuthzPolicyEnforcementTest.java:383]: Need to fail here if no exception

Comment (lmsurpre) [reply fhir-smart/src/test/java/org/linuxforhealth/fhir/smart/test/AuthzPolicyEnforcementTest.java:383]: fixed line 378 to use a different reosurce type and added the fail call

**Review by punktilious — APPROVED** [review]
LGTM

**Pull request merged**